### ISSUE TITLE
[naming] Promote _LRScheduler to LRScheduler

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#23957]
Given that it's public API and a first class concept, I think it would be nice to remove the internal-implementation-detail-suggesting underscore from the name (e.g. Optimizer has no underscore)

cc @vincentqb

**Comment by ezyang** [comment]
cc @vincentqb

**Comment by vincentqb** [comment]
Thanks for the input! This ties in with #22343, and the ongoing redesign of Schedulers. It may be a good idea to get rid of the _LRScheduler and replace directly by a general Scheduler that allows us to choose what to schedule (e.g. learning rate, weights, or other parameters). Thoughts?

**Comment by vadimkantorov** [comment]
Also some schedulers are formulated as a function of iteration (and not epoch index). E.g. this can influence renaming `last_epoch` field in _LRScheduler to `last_iter`.

**Comment by gchanan** [comment]
@vincentqb can this be closed now?

**Comment by vincentqb** [comment]
Since this could encourage people to experiment with the learning rate scheduler class, I'm ok if @vadimkantorov wants to open a pull request promoting `_LRScheduler` to `LRScheduler`, and keeping the old name for backward compatibility, potentially with a deprecation warning.

**Comment by vadimkantorov** [comment]
GIven all my problems with current LR schedulers design, I don't want any more to champion this issue ^^

**Comment by vadimkantorov** [comment]
Seems implemented in https://github.com/pytorch/pytorch/pull/88503
